### DISCUSSION TITLE
How to detect if something is within a MenuTrigger

### DISCUSSION THREAD

**Discussion by synecdokey** [adobe/react-spectrum#6345] [Q&A] [answered]
We styled a button differently depending on whether it is part of a MenuTrigger. We did so by using within the button to 'detect' it:
```tsx
const [{ id: menuId }] = useContextProps<
  MenuProps<MenuItemProps>,
  MenuProps<MenuItemProps>,
  HTMLDivElement
>({}, null, MenuContext);
```

Am I missing a more obvious way to do this?

**Comment by snowystinger** [comment, marked as answer]
Could you make your own context which you write to in a menu trigger and read from in a button?

**Reply by synecdokey** [reply]
That's actually so simple I'm kind of ashamed I didn't think about this. I'll see if that fixes my issues. Thanks!

**Reply by synecdokey** [reply]
This solved the question nicely, though I ended up with a simpler solution: `useContext(MenuContext)`, just checking if it is null is enough of a hint.
